Lanchonete/Lanchonete/Model/Sandwich.swift

```diff
@@ -5,28 +5,50 @@
 //  Created by vera.dias on 1/6/21.
 //
 
-import Foundation
+import UIKit
 public enum SandwichType:  String, CaseIterable {
+    
     case xBacon = "X-Bacon"
     case xEgg = "X-Egg"
     case xBurguer = "X-Burguer"
     case xEggBacon = "X-EggBacon"
     
-    var ingredientsInSandwich: [IngredientInSandwich] {
+    var ingredientsInSandwich: [Ingredient] {
+        switch self {
+        case .xBacon:
+            return [.hamburguer,.cheese,.bacon]
+        case .xBurguer:
+            return [.hamburguer,.cheese]
+        case .xEgg:
+            return [.hamburguer,.cheese,.egg]
+        case .xEggBacon:
+            return [.hamburguer,.cheese,.egg,.bacon]
+        }
+    }
+    
+    var name: String {
         switch self {
         case .xBacon:
-            return [IngredientInSandwich(ingredient: .hamburguer, quantity: 1),
-                    IngredientInSandwich(ingredient: .cheese, quantity: 1),
-                    IngredientInSandwich(ingredient: .bacon, quantity: 1)]
+            return "X-Bacon"
         case .xBurguer:
-            return [IngredientInSandwich(ingredient: .hamburguer, quantity: 1), IngredientInSandwich(ingredient: .cheese, quantity: 1)]
+            return "X-Burguer"
         case .xEgg:
-            return [IngredientInSandwich(ingredient: .hamburguer, quantity:1),IngredientInSandwich(ingredient: .cheese, quantity: 1),IngredientInSandwich(ingredient: .egg, quantity: 1)]
+            return "X-Egg"
         case .xEggBacon:
-            return [IngredientInSandwich(ingredient: .egg, quantity: 1),
-                    IngredientInSandwich(ingredient: .bacon, quantity: 1),
-                    IngredientInSandwich(ingredient: .hamburguer, quantity: 1),
-                    IngredientInSandwich(ingredient:.cheese, quantity: 1)]
+            return "X-EggBacon"
+        }
+    }
+    
+    var image: UIImage {
```

Comment:
In the same spirit of the other comments, I would suggest to return something that does not depend on UIKit, like an enum or a protocol, for instance

Reply:
I also refactored this, it now returns a "code" aka name of the image 😉